zebrad/tests/acceptance.rs

```diff
@@ -1455,7 +1455,73 @@ async fn tracing_endpoint() -> Result<()> {
     Ok(())
 }
 
-// TODO: RPC endpoint and port conflict tests (#3165)
+#[tokio::test]
+async fn rpc_endpoint() -> Result<()> {
+    use hyper::{body::to_bytes, Body, Client, Method, Request};
+    use serde_json::Value;
+
+    zebra_test::init();
+
+    // [Note on port conflict](#Note on port conflict)
+    let port = random_known_port();
+    let endpoint = format!("127.0.0.1:{}", port);
+    let url = format!("http://{}", endpoint);
+
+    // Write a configuration that has RPC listen_addr set
+    let mut config = default_test_config()?;
+    config.rpc.listen_addr = Some(endpoint.parse().unwrap());
+
+    let dir = testdir()?.with_config(&mut config)?;
+    let child = dir.spawn_child(&["start"])?;
+
+    // Run `zebrad` for a few seconds before testing the endpoint
+    // Since we're an async function, we have to use a sleep future, not thread sleep.
+    tokio::time::sleep(LAUNCH_DELAY).await;
```

Comment:
It looks like the test is failing because Zebra takes more than 10 seconds to start.

You could try waiting for the port to open instead?
https://github.com/ZcashFoundation/zebra/blob/397ba1fef73addc5a543a164daa1ce56bdcdeac1/zebrad/tests/acceptance.rs#L1761-L1765

Reply:
Yea, this should help. Added it at https://github.com/ZcashFoundation/zebra/pull/3641/commits/b38910834a39ba78cd972e8da83c4901cba612e7